geotrellis-sentinelhub/src/main/scala/org/openeo/geotrellissentinelhub/CatalogApi.scala

```diff
@@ -77,6 +80,50 @@ object DefaultCatalogApi {
     extends JsonFeatureCollection(features)
   private case class PagedJsonFeatureCollectionMap(features: List[Json], context: PagingContext)
     extends JsonFeatureCollectionMap(features)
+
+
+  // This helper function is called below to grab the ID field for Map keys
+  private def getFeatureID(js: Json): String = {
+    val cursor = js.hcursor
+    val id = cursor.downField("id")
+    id.as[String] match {
+      case Right(i) => i
+      case _ =>
+        id.as[Int] match {
+          case Right(i) => i.toString
+          case _ => throw new RuntimeException("Feature expected to have \"ID\" field" + cursor.history)
+        }
```

Comment:
Pattern matching on Either can be replaced with `valueOr(throw new ...)`.

Reply:
Fixed in https://github.com/Open-EO/openeo-geotrellis-extensions/pull/134